### PR TITLE
CORS-2994: AWS BYO IAM profile implementation

### PR DESCRIPTION
**Pull request by r4f4**
These steps will be needed to test the BYO IAM profile feature.

Support is being added to the Installer via https://github.com/openshift/installer/pull/8689

### CODE REVIEW

**Comment by r4f4** [comment]
The installer will ignore the `iamProfile` field until the installer PR merges, but at least we can validate whether the profiles are created/destroyed correctly:

/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-custom-iam-profile

**Comment by r4f4** [comment]
IAM profiles created: [logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/54841/rehearse-54841-pull-ci-openshift-installer-master-e2e-aws-ovn-custom-iam-profile/1816579953920053248/artifacts/e2e-aws-ovn-custom-iam-profile/aws-provision-iam-profile/build-log.txt)
IAM profiles set in [install-config.yaml](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/54841/rehearse-54841-pull-ci-openshift-installer-master-e2e-aws-ovn-custom-iam-profile/1816579953920053248/artifacts/e2e-aws-ovn-custom-iam-profile/ipi-install-install/artifacts/install-config.yaml)
IAM profiles destroyed:  [logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/54841/rehearse-54841-pull-ci-openshift-installer-master-e2e-aws-ovn-custom-iam-profile/1816579953920053248/artifacts/e2e-aws-ovn-custom-iam-profile/aws-deprovision-iam-profile/build-log.txt)

Just need to investigate why this is happening:
```
usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

aws: error: argument --instance-profile-name: expected one argument
```

**Comment by r4f4** [comment]
> Just need to investigate why this is happening:
> 
> ```
> usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
> To see help text, you can run:
> 
>   aws help
>   aws <command> help
>   aws <command> <subcommand> help
> 
> aws: error: argument --instance-profile-name: expected one argument
> ```

Oh of course! Because we need https://github.com/openshift/installer/pull/8689 for the Installer to actually use the existing profile, it ignored that and created its own IAM profile which was destroyed together with the cluster. Therefore when we tried to [list instances to get the profiles](https://github.com/openshift/release/pull/54841/files#diff-168d4eaf4b2ffc8538493f20c91cd8115d0c8a5f95132b1b293761b76fc4df56R92) they were already gone and `${master_profile}` is empty.

File: ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-ref.metadata.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+	"path": "aws/deprovision/iam-profile/aws-deprovision-iam-profile-ref.yaml",
+	"owners": {
+		"approvers": [
```
Comment (barbacbd) [inline ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-ref.metadata.json:4]: do you want to add yourself?

File: ci-operator/step-registry/aws/provision/iam-profile/aws-provision-iam-profile-ref.metadata.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+	"path": "aws/provision/iam-profile/aws-provision-iam-profile-ref.yaml",
+	"owners": {
+		"approvers": [
+			"patrickdillon",
```
Comment (barbacbd) [inline ci-operator/step-registry/aws/provision/iam-profile/aws-provision-iam-profile-ref.metadata.json:5]: Same here.

File: ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.metadata.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+	"path": "ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.yaml",
+	"owners": {
+		"approvers": [
+			"patrickdillon",
```
Comment (barbacbd) [inline ci-operator/step-registry/ipi/conf/aws/byo-iam-profile/ipi-conf-aws-byo-iam-profile-ref.metadata.json:5]: Same here.

File: ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-ref.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+ref:
+  as: aws-deprovision-iam-profile
+  from_image:
+    namespace: ocp
+    name: "4.12"
```
Comment (barbacbd) [inline ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-ref.yaml:5]: Can/should we use a more updated image?

Comment (r4f4) [reply ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-ref.yaml:5]: I guess we can but I wanted to keep it consistent with the iam-role step.

Comment (yunjiang29) [reply ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-ref.yaml:5]: The only tool we require in this image is `yq-go` (and `aws`), so 4.12 is good for us, we can use an updated image for both byo-role/profile, but it's not necessary.

**Comment by yunjiang29** [comment]
@r4f4 This PR looks good to me, let me know when it's ready for merge (after https://github.com/openshift/installer/pull/8689 got merged)

**Comment by r4f4** [comment]
@yunjiang29 I was thinking of merging this before https://github.com/openshift/installer/pull/8689 and then use the installer presubmit introduced here in the installer PR. WDYT?

**Comment by yunjiang29** [comment]
@r4f4 Sure.

/lgtm

**Pull request merged**